table/block_based/block.h

```diff
@@ -405,11 +390,23 @@ class BlockIter : public InternalIteratorBase<TValue> {
   template <typename DecodeEntryFunc>
   inline bool ParseNextKey(bool* is_shared);
 
-  InternalKeyComparator icmp() {
-    return InternalKeyComparator(raw_ucmp_, false /* named */);
-  }
+  void InitializeBase(const Comparator* raw_ucmp, const char* data,
+                      uint32_t restarts, uint32_t num_restarts,
+                      SequenceNumber global_seqno, bool block_contents_pinned) {
+    assert(data_ == nullptr);  // Ensure it is called only once
+    assert(num_restarts > 0);  // Ensure the param is valid
 
-  UserComparatorWrapper ucmp() { return UserComparatorWrapper(raw_ucmp_); }
+    icmp_ =
+        std::make_unique<InternalKeyComparator>(raw_ucmp, false /* named */);
```

Comment:
Can `InternalKeyComparator` wrap a `UserComparatorWrapper`?

Reply:
It seems like InternalKeyComparator already has a UserComparatorWrapper as a private member. Calling icmp_->user_comparator() refers to it.